R/xportr.R

```diff
@@ -0,0 +1,71 @@
+#' Wrapper to apply all core xportr functions and write xpt
+#'
+#' @param .df A data frame of CDISC standard.
+#' @param var_metadata A data frame containing variable level metadata
+#' @param domain Appropriate CDSIC dataset name, e.g. ADAE, DM. Used to subset
+#'   the metadata object. If none is passed, then name of the dataset passed as
+#'   .df will be used.
+#' @param verbose The action this function takes when an action is taken on the
+#'   dataset or function validation finds an issue. See 'Messaging' section for
+#'   details. Options are 'stop', 'warn', 'message', and 'none'
+#' @param df_metadata A data frame containing dataset level metadata.
+#' @param path Path where transport file will be written. File name sans will be
+#'   used as `xpt` name.
+#' @param strict_checks If TRUE, xpt validation will report errors and not write
+#'   out the dataset. If FALSE, xpt validation will report warnings and continue
+#'   with writing out the dataset. Defaults to FALSE
```

Comment:
2 comments about cleaning duplicate documentation

1. Have `xportr_length` inherit from this function and add any missing parameter there.
    - all other functions continue to inherit from `xportr_length`
1. Keep using `metadata` on `xportr()` and migrate `xportr_df_label` parameter name to `df_metadata`
    - This would allow all functions to inherit from this one and have standard names across functions
    - With a deprecation process in df_label

Reply:
@averissimo thanks for the review! I'm a little unclear on what you're suggesting here, but I like the idea of reducing duplicate docs. Are you saying `xportr()` should have a `metadata` and a `df_metadata` parameter, and then that all other functions should follow that pattern?

Reply:
Yes, that it. I'll rephrase it though:

The idea is to reduce duplicate parameter documentation. Currently all the `xportr_*` functions inherit the parameter information from `xportr_length` with:

```r
#' ...
#' @inheritParams xportr_length
#' ...
xportr_format <- function(...)
```

Ideally, I think all should inherit from `#' @inheritParams xportr`, but given that it contains `var_metadata` and `df_metadata` (that don't map to any other) it needs some intermediate action.

Note: if we simply add `#' @param metadata ...` to `xportr()` it will add that parameter in the man page

![image](https://github.com/atorus-research/xportr/assets/211358/1685bcca-e5f5-48a3-a40f-48bc713b78c1)

#### Possible Actions (one of)

1. Keep using `xportr_length` as the template for all other `xportr_*` functions
    - Have `xportr_length` inherit the majority of the params from `xportr()` with `#' @inheritParams xportr`
    - Add the missing `metadata` parameter that doesn't exist on `xportr`
1. Rename `var_metadata` parameters in `xportr` to `metadata`
    - All `xportr_*` function will inherit from `xportr()` with `#' @inheritParams xportr`

Optionally, we could rename the parameter on `xportr_df_label` from `metadata` to `df_metadata` to have a consistent naming convention. _(but this is not really required as `xportr_df_label` overwrites the `metadata` parameter documentation)_

Reply:
Okay, I think I got option 1 done, let me know what you think. Does this method leave extra parameters in other functions docs though? Like will `length` get `df_metadata` and `var_metadata` included as parameters?